L1Trigger/L1THGCal/interface/be_algorithms/HGCalTowerMap2DImpl.h

```diff
@@ -0,0 +1,53 @@
+#ifndef __L1Trigger_L1THGCal_HGCalTowerMap2DImpl_h__
+#define __L1Trigger_L1THGCal_HGCalTowerMap2DImpl_h__
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include "DataFormats/L1THGCal/interface/HGCalTriggerCell.h"
+#include "DataFormats/L1THGCal/interface/HGCalTowerMap.h"
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerTools.h"
+
+
+class HGCalTowerMap2DImpl{
+
+ public:
+
+  HGCalTowerMap2DImpl( const edm::ParameterSet &conf); 
+  
+  void resetTowerMaps( );
+
+  void buildTowerMap2D( const std::vector<edm::Ptr<l1t::HGCalTriggerCell>> & triggerCellsPtrs,
+			l1t::HGCalTowerMapBxCollection & towermaps
+			);
+
+
+  void eventSetup(const edm::EventSetup& es) 
+    {
+        triggerTools_.setEventSetup(es);
+    }
+
+  
+ private:
+
+  static const int kLayersEE_ = 28;
+  static const int kLayersFH_ = 12;  
+  static const unsigned kLayersBH_ = 12;
+  static const unsigned kLayers_ = kLayersEE_+kLayersFH_+kLayersBH_;
+  std::vector<l1t::HGCalTowerMap> towerMaps_; //towerMaps for each HGC layer
```

Comment:
`towerMaps_` is used only in `buildTowerMap2D()` and rebuilt each time this function is called, so it could be a local variable, it doesn't need to be a class member.